### PR TITLE
Mark Tmds.DBus.* as PrivateAssets in Avalonia.FreeDesktop.csproj

### PR DESCRIPTION
**Pull request by robloo**
## What does the pull request do?

See discussion here: https://github.com/AvaloniaUI/Avalonia/discussions/10871#discussioncomment-5494996

### CODE REVIEW

File: src/Avalonia.FreeDesktop/Avalonia.FreeDesktop.csproj [hunk 1]
```diff
@@ -12,8 +12,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Tmds.DBus.Protocol" Version="0.14.0" />
-    <PackageReference Include="Tmds.DBus.SourceGenerator" Version="0.0.5" />
+    <PackageReference Include="Tmds.DBus.Protocol" Version="0.14.0" PrivateAssets="All" />
```
Comment (robloo) [inline src/Avalonia.FreeDesktop/Avalonia.FreeDesktop.csproj:15]: @kekekeks I also marked `Tmds.DBus.Protocol` with `PrivateAssets="All"`. If that is incorrect please let me know and I'll fix it.

**Comment by robloo** [comment]
Well, turns out this was used by a few other Linux projects and now the build is broken.

**Comment by jp2masa** [comment]
I think you can't mark `Tmds.DBus.Protocol` with `PrivateAssets="all"` because it's a runtime dependency. Source generator is fine because it's only a compile time dependency.

File: src/Avalonia.FreeDesktop/Avalonia.FreeDesktop.csproj (hunk 1 above)
Comment (robloo) [reply src/Avalonia.FreeDesktop/Avalonia.FreeDesktop.csproj]: Nevermind: Had to revert this part because it didn't build. See comment below.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032704-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Pull request merged**